pkg/resources/network_policy.go

```diff
@@ -20,19 +22,31 @@ var networkPolicySchema = map[string]*schema.Schema{
 		Description: "Specifies the identifier for the network policy; must be unique for the account in which the network policy is created.",
 		ForceNew:    true,
 	},
+	"allowed_network_rule_list": {
+		Type:        schema.TypeSet,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+		Optional:    true,
+		Description: "Specifies a list of fully qualified network rules that contain the network identifiers that are allowed access to Snowflake.",
+	},
+	"blocked_network_rule_list": {
+		Type:        schema.TypeSet,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+		Optional:    true,
+		Description: "Specifies a list of fully qualified network rules that contain the network identifiers that are denied access to Snowflake.",
+	},
```

Comment:
I am not sure, what the best way to solve a problem here. Network rules are schema level objects, thus it is necessary to pass a list of fully qualified names as parameters. The problem is the validation. There is a IsValidIdentifier function, which does such a check, however, it is only defined for collections. In addition, the entire framework does not to use ValidateDiagFunc functions for collections (not implemented).

We could leave this at it is, however, the provider crashes with an error, which is not easily understood, when someone passes e.g. "foo" as a network rule name. Therefore, there should be a form of validation.

I am thinking about adding a custom CustomizeDiff function into the schema.Resource struct, which does the validation. For this I would extract the main validation logic from the IsValidIdentifier function into a function, which can be used outside the ValidateDiagFunc  mechanism and use it to check very entry of the network rule list.

Does this work for you or would you prefer another solution?

Reply:
Hey 👋 
I think the proposed CustomizeDiff would work fine, but please add an acceptance test showing it throws an expected error whenever the incorrect identifier is passed.

Reply:
Hi, thanks for you contribution :) What do you mean by `There is a IsValidIdentifier function, which does such a check, however, it is only defined for collections`? I can see that it supports singular ID types. I think you can use `parseNetworkRulesList` which essentially converts `schema.TypeSet` -> `[]sdk.SchemaObjectIdentifier` an call `IsValidIdentifier` on each element in the returned slice, right? Correct me if I'm missing something. cc @sfc-gh-jcieslak

Reply:
@sfc-gh-jmichalak 
Regarding `There is a IsValidIdentifier function, which does such a check, however, it is only defined for collections`: There was an error on my side, I actually meant `There is a IsValidIdentifier function, which does such a check, however, it is only defined for single elements and not for collections`. Sorry for the confusion made.

@sfc-gh-jcieslak 
You're suggestion worked perfectly! Then I think I am finally finished with the actual implementation. I will let you know when I have added the tests and finalized the docs.

Reply:
@sfc-gh-jcieslak @sfc-gh-jmichalak 
I have added tests and docs, I hope that I haven't overlooked something. A review of the PR would be greatly appreciated :)